### PR TITLE
simple header-based pagination integrated into from_pools

### PR DESCRIPTION
**Pull request by JTCunning**
Once we discuss the code completion, I'll throw some tests in.

### CODE REVIEW

File: clustoapi/server.py [hunk 1]
```diff
@@ -306,15 +306,25 @@ def get_from_pools():
     drivers = bottle.request.params.getall('driver')
     children = bottle.request.params.get('children', default=True, type=bool)
     mode = bottle.request.headers.get('Clusto-Mode', default='compact')
+    current = int(bottle.request.headers.get('Clusto-Page', default='0'))
```
Comment (JTCunning) [inline clustoapi/server.py]: Looking for a more elegant way to cast this.

File: clustoapi/server.py [hunk 2]
```diff
@@ -306,15 +306,25 @@ def get_from_pools():
...
+    per = int(bottle.request.headers.get('Clusto-Per-Page', default='50'))
 
     try:
         ents = clusto.get_from_pools(
             pools, clusto_types=types, clusto_drivers=drivers, search_children=children
         )
         results = []
+        headers = {}
+        if current:
+            ents, total = util.page(list(ents), current=current, per=per)
```
Comment (JTCunning) [inline clustoapi/server.py]: ``from_pools`` returns a set and my list math didn't like that.

Comment (JTCunning) [inline clustoapi/server.py]: Additionally, ignoring the second unpack (like if I took out ``total``) turns the returned object into an actual tuple, is there any way to combat that in the method and return just what is requested to be unpacked? Iono.

**Comment by motivator** [comment]
sure. looks ok.

**Comment by thekad** [comment]
I feel like the default pagination should be configurable, so we can accomplish that suggestion about http://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#pagination
Of course we can just skip the config thing and make it page by default at 50, and still do the Link header work...

**Comment by thekad** [comment]
Nevermind, was confoose

**Comment by thekad** [comment]
I am :+1: as long as we document the caveat that this pagination only works for *presentation*, meaning you still fetch the full recordset from clusto, but only spit out a fraction of it to the http client.

**Comment by JTCunning** [comment]
I have some wonky branch stuff going on. Closing this in favor of https://github.com/clusto/clusto-apiserver/pull/6.

**Pull request closed without merging**